components/camel-platform-http-vertx/src/main/java/org/apache/camel/component/platform/http/vertx/VertxPlatformHttpConsumer.java

```diff
@@ -163,6 +146,63 @@ private String configureEndpointPath(PlatformHttpEndpoint endpoint) {
         return PATH_PARAMETER_PATTERN.matcher(path).replaceAll(":$1");
     }
 
+    private void handleRequest(RoutingContext ctx) {
+        final Vertx vertx = ctx.vertx();
+        final Exchange exchange = toExchange(ctx);
+
+        //
+        // We do not know if any of the processing logic of the route is synchronous or not so we
+        // need to process the request on a thread on the Vert.x worker pool.
+        //
+        // As example, assuming the platform-http component is configured as the transport provider
+        // for the rest dsl, then the following code may result in a blocking operation that could
+        // block Vert.x event-loop for too long if the target service takes long to respond, as
+        // example in case the service is a knative service scaled to zero that could take some time
+        // to be come available:
+        //
+        //     rest("/results")
+        //         .get("/{id}")
+        //         .route()
+        //             .removeHeaders("*", "CamelHttpPath")
+        //             .to("rest:get:?bridgeEndpoint=true");
+        //
+        vertx.executeBlocking(
+                promise -> {
+                    try {
+                        createUoW(exchange);
+
+                        // no need to use an async processor as the processing happen in
+                        // a dedicated thread ans it won't block the Vert.x event loop
+                        getProcessor().process(exchange);
```

Comment:
You should favour async processor anyway as that makes camel use it where it can

Reply:
@davsclaus something like: `getAsyncProcessor().process(exchange, c -> promise.complete());` ?

Reply:
Yes exactly

Reply:
or maybe better 

```
getAsyncProcessor().process(exchange, c -> {
    if (!exchange.isFailed()) {
        promise.complete();
    } else {
        promise.fail(exchange.getException());
    }
});
```

Reply:
@davsclaus ^

Reply:
Yeah just be sure that when promise.fail will also trigger the result block (I dont know what it does) as the UoW must be done as it does below in the finally block